art/step/step_savers.py

```diff
@@ -7,28 +7,26 @@
 
 class StepSaver(ABC):
     @abstractmethod
-    def save(self, obj: any, step_id: int, step_name: str, filename: str):
+    def save(self, obj: any, step_id: str, step_name: str, filename: str):
         pass
 
     @abstractmethod
     def load(self, step_name: str, filename: str):
         pass
 
-    def ensure_directory(self, step_id: int, step_name: str):
-        return (BASE_PATH / f"{step_id}_{step_name}").mkdir(parents=True, exist_ok=True)
+    def ensure_directory(self, step_id: str, step_name: str):
+        return (BASE_PATH / f"{step_name}_{step_id}").mkdir(parents=True, exist_ok=True)
```

Comment:
step_id was before name on purpose so that files would be sorted for user

Reply:
Sounds reasonable, I'll change that